tensorboard/components/vz_chart_helpers/vz-chart-tooltip.ts

```diff
@@ -90,19 +90,43 @@ Polymer({
   updateAndPosition(anchorNode: Element, newDom: Element[]) {
     newDom.forEach(row => this.scopeSubtree(row));
 
+    if (!anchorNode.isConnected) {
+      throw new DOMError('AnchorNode must be connected/mounted');
+    }
+
     window.cancelAnimationFrame(this._raf);
     this._raf = window.requestAnimationFrame(() => {
       if (!this.isAttached) return;
       this._repositionImpl(anchorNode);
     });
   },
 
+  /**
+   * Finds the scrollable container using heuristics - using computed style,
+   * scrollHeight, and clientHeight. In case scrollHeight is greater, it means
+   * that the container can show scorllbar and the anchor node can be hidden.
+   * This is by no means only way a content is clipped by other elements but,
+   * for use cases so far, this is good enough.
+   */
+  _getScrollableContainer(node: Element): Element {
+    while (node && node !== document.body) {
```

Comment:
> > When can the node become null before reaching `document.body`?
>
> It can be fasly/null when a node is not mounted to the DOM.

Under what circumstances do we try to get the scrollable container of a
node that’s not mounted to the DOM? Why is returning `document.body` the
right thing to do in that case?

Reply:
May I ask what you are implicitly asking here? Are you implying I should return null?

In either case, I have changed the code to add explicit input validation on the public API side.